CleanTalk/XF/Repository/Option.php

```diff
@@ -12,18 +12,50 @@ class Option extends XFCP_Option
 {
 	public function updateOptions(array $values)
 	{
-		$options = parent::updateOptions($values);
-
+		parent::updateOptions($values);
 		$plugin_version = $this->app()->addOnManager()->getById('CleanTalk')->getJsonVersion();
-		if (isset($values['ct_apikey']) && $values['ct_apikey'] != '')
-		{
-			CleantalkAPI::method__send_empty_feedback($values['ct_apikey'], 'xenforo2-' . $plugin_version['version_id']);
-			
-			if (isset($values['ct_sfw']) && intval($values['ct_sfw']) == 1)
-			{	
-				CleantalkFuncs::apbct_sfw_update($values['ct_apikey']);
-				CleantalkFuncs::apbct_sfw_send_logs($values['ct_apikey']);		
-			}
-		}
-	}	
+
+        if ($this->checkAccessKey($values['ct_apikey']))
+        {
+            CleantalkAPI::method__send_empty_feedback($values['ct_apikey'], 'xenforo2-' . $plugin_version['version_id']);
+
+            if (isset($values['ct_sfw']) && intval($values['ct_sfw']) == 1)
+            {
+                CleantalkFuncs::apbct_sfw_update($values['ct_apikey']);
+                CleantalkFuncs::apbct_sfw_send_logs($values['ct_apikey']);
+            }
+        }
+    }
+
+    /**
+     * Check the key and set an error on options if is incorrect.
+     * @param $ct_apikey
+     * @return bool
+     */
+    private function checkAccessKey($ct_apikey){
```

Comment:
or **validateApiKey**

Reply:
Thought about that. In all the places in the code, the word is like an "access key". To prevent misreading I decided to keep the word form as is. In another way, we need to refactor.

Reply:
Furthermore, Xenforo 2 has an API key variable for its own license validation.

Reply:
However, some refactoring were applied here: https://github.com/CleanTalk/xenforo2-antispam/commit/865e1a90ad84a3d2e08065088bffe2dc455a9dc9